checker/src/main/java/org/checkerframework/checker/determinism/DeterminismQualifierPolymorphism.java

```diff
@@ -80,25 +83,56 @@ protected void replace(
                     qualHierarchy.leastUpperBounds(replacementsPolyDet, replacementsPolyDetNoOND);
             replacements = new AnnotationMirrorSet(lub);
         }
+        if (replacements == null) {
+            return;
+        }
 
         switch (value) {
             case "":
                 type.replaceAnnotations(replacements);
                 return;
             case "use":
-                if (!replacements.contains(factory.POLYDET)
-                        && !replacements.contains(factory.POLYDET_UP)
-                        && !replacements.contains(factory.POLYDET_DOWN)
-                        && !replacements.contains(factory.POLYDET_UPDET)) {
-                    type.replaceAnnotations(replacements);
+                if (replacementsPolyDet != null
+                        && replacementsPolyDet.contains(factory.POLYDET_NOORDERNONDET)) {
+                    type.replaceAnnotation(factory.POLYDET_USENOORDERNONDET);
```

Comment:
I looks like this annotation is replaced again in the next if statement.  So what is the purpose of this?